client/reader/stream/post-lifecycle.jsx

```diff
@@ -51,6 +54,39 @@ export default class PostLifecycle extends React.PureComponent {
 		this.updatePost( nextProps );
 	}
 
+	shouldComponentUpdate( nextProps, nextState ) {
+		const {
+			isSelected, suppressSiteNameLink, showPostHeader, showFollowInHeader,
+			showPrimaryFollowButtonOnCards, showSiteName, isDiscoverStream,
+		} = this.props;
+
+		const shouldUpdate = this.state.post !== nextState.post ||
```

Comment:
if you're checking everything, you could either use a shallow compare, or inherit from React.PureComponent. Or is this not the full set of props?